devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1035,6 +1163,374 @@ def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejec
+				def startDay = startDateTime.day
+				def startMonth = startDateTime.month
+				def startYear = startDateTime.year.toString()
+				startYear = startYear.substring(2, startYear.length())
+				startYear = Integer.parseInt(startYear, 10)
+	
+				def endDateTime = getYearDayTimeDetails(endDate, endTime)
+				def stopHour = endDateTime.hour
+				def stopMinute = endDateTime.minute
+				def stopDay = endDateTime.day
+				def stopMonth = endDateTime.month
+				def stopYear = endDateTime.year.toString()
+				stopYear = stopYear.substring(2, stopYear.length())
+				stopYear = Integer.parseInt(stopYear, 10)
+	
+				if (cmd.startYear == startYear && cmd.startMonth == startMonth && cmd.startDay == startDay && cmd.startHour == startHour && cmd.startMinute == startMinute &&
+						cmd.stopYear == stopYear && cmd.stopMonth == stopMonth && cmd.stopDay == stopDay && cmd.stopHour == stopHour && cmd.stopMinute == stopMinute) {
+					log.trace "[DTH] ScheduleEntryLockYearDayReport() -  Schedule set successfully for codeID := $codeID"
+					result << scheduleSetEvent(codeID, SCHEDULE_TYPE_YEAR_DAY)
+					def deviceName = device.displayName
+					def map = [name: "scheduleChanged", value: "$codeID $SCHEDULE_TYPE_YEAR_DAY $changeType", descriptionText: "${getStatusForDescription(changeType)} limited schedule for \"$codeName\"", isStateChange: true, displayed: true]
+					map.data = [lockName: deviceName, notify: true, notificationText: "${getStatusForDescription(changeType)} \"$codeName\" with limited schedule in $deviceName at ${location.name}"]
+					result << createEvent(map)
+				} else {
+					log.trace "[DTH] ScheduleEntryLockYearDayReport() -  Schedule set failed for codeID := $codeID"
+					result << createEvent(name: "scheduleChanged", value: "$codeID $SCHEDULE_TYPE_YEAR_DAY failed", descriptionText: "Limited schedule set failed for \"$codeName\"", isStateChange: true, displayed: false)
+				}
+			} else {
+				log.trace "[DTH] ScheduleEntryLockYearDayReport() -  Schedule found while scanning for codeID := $codeID"
+				def startYear = (cmd.startYear < 10) ? "0${cmd.startYear}" : "${cmd.startYear}"
+				def startMonth = (cmd.startMonth < 10) ? "0${cmd.startMonth}" : "${cmd.startMonth}"
+				def startDay = (cmd.startDay < 10) ? "0${cmd.startDay}" : "${cmd.startDay}"
+				def startHour = (cmd.startHour < 10) ? "0${cmd.startHour}" : "${cmd.startHour}"
+				def startMinute = (cmd.startMinute < 10) ? "0${cmd.startMinute}" : "${cmd.startMinute}"
+				def stopYear = (cmd.stopYear < 10) ? "0${cmd.stopYear}" : "${cmd.stopYear}"
+				def stopMonth = (cmd.stopMonth < 10) ? "0${cmd.stopMonth}" : "${cmd.stopMonth}"
+				def stopDay = (cmd.stopDay < 10) ? "0${cmd.stopDay}" : "${cmd.stopDay}"
+				def stopHour = (cmd.stopHour < 10) ? "0${cmd.stopHour}" : "${cmd.stopHour}"
+				def stopMinute = (cmd.stopMinute < 10) ? "0${cmd.stopMinute}" : "${cmd.stopMinute}"
+				state["setYearDayStartDate$codeID"] = "20${startYear}-${startMonth}-${startDay}"
```

Comment:
You could save a lot of lines using:

`sprintf("%02d", variableName)`

Reply:
or in this case specifically:

`sprintf("20%02d-%02d-%02d", startYear, startMonth, startDay)`